### PR TITLE
[runtime] Avoid regaining the lock in the Monitor.Wait () icall when …

### PR DESCRIPTION
**Pull request by vargaz**
…the thread is in StopRequested state, exit eagerly instead. Fixes #41644.

### CODE REVIEW

File: mono/metadata/monitor.c [hunk 1]
```diff
@@ -1305,6 +1305,18 @@ ves_icall_System_Threading_Monitor_Monitor_wait (MonoObject *obj, guint32 ms)
 	 */
 	mono_thread_clr_state (thread, ThreadState_WaitSleepJoin);
 
+	if (mono_thread_test_state (thread, ThreadState_StopRequested)) {
+		/*
```
Comment (kumpera) [inline mono/metadata/monitor.c]: This looks wrong in many levels.

Comment (kumpera) [inline mono/metadata/monitor.c]: Let me research the problem a bit.

**Comment by vargaz** [comment]
An alternative would be collecting the held monitors in a list in MonoInternalThread.

**Comment by vargaz** [comment]
Fixed it another way in f5ba317644dc3d41093495f91a29e024044e6b18.

**Pull request closed without merging**